diff --git a/post-processor/vagrant-cloud/client.go b/post-processor/vagrant-cloud/client.go
index a66d3eb..5f107ee 100644
--- a/post-processor/vagrant-cloud/client.go
+++ b/post-processor/vagrant-cloud/client.go
@@ -125,6 +125,7 @@ func (v VagrantCloudClient) Upload(path string, url string) (*http.Response, err
 	log.Printf("Post-Processor Vagrant Cloud API Upload: %s %s", path, url)
 
 	request.ContentLength = fi.Size()
+	v.client.Timeout = 300 * time.Second
 	resp, err := v.client.Do(request)
 
 	log.Printf("Post-Processor Vagrant Cloud Upload Response: \n\n%+v", resp)
